File: src/models/Patient.ts
import mongoose, { Schema, Model } from 'mongoose';
import { IPatient } from '../types';

const patientSchema = new Schema<IPatient>(
    {
        userId: {
            type: String,
            required: true,
            ref: 'User',
        },
        patientId: {
            type: String,
            required: true,
            unique: true,
        },
        name: {
            type: String,
            required: true,
        },
        email: {
            type: String,
            default: '',
        },
        phone: {
            type: String,
            required: true,
        },
        dob: {
            type: Date,
            required: true,
        },
        bloodGroup: {
            type: String,
            default: '',
        },
        address: {
            type: String,
            default: '',
        },
        imageUrl: {
            type: String,
            default: '',
        },
        qrCodeUrl: {
            type: String,
            default: '',
        },
    },
    {
        timestamps: true,
    }
);

// Create indexes
patientSchema.index({ userId: 1 });
patientSchema.index({ patientId: 1 });
patientSchema.index({ name: 'text' }); // For text search

export const Patient: Model<IPatient> = mongoose.model<IPatient>('Patient', patientSchema);

File: src/main/java/com/medicare/backend/models/Patient.java
package com.medicare.backend.models;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.index.TextIndexed;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;

import java.time.LocalDateTime;
import java.util.Date;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "patients")
public class Patient {
    @Id
    private String id;

    @Indexed
    private String userId;

    @Indexed(unique = true)
    private String patientId;

    @TextIndexed
    private String name;

    private String email;

    private String phone;

    private Date dob;

    private String bloodGroup;

    private String address;

    private String imageUrl;

    private String qrCodeUrl;

    @CreatedDate
    private LocalDateTime createdAt;

    @LastModifiedDate
    private LocalDateTime updatedAt;
}
